xesmf/frontend.py

```diff
@@ -72,9 +72,45 @@ def ds_to_ESMFgrid(ds, need_bounds=False, periodic=None, append=None):
     return grid, lon.shape
 
 
+def ds_to_ESMFlocstream(ds):
+    '''
+    Convert xarray DataSet or dictionary to ESMF.LocStream object.
+
+    Parameters
+    ----------
+    ds : xarray DataSet or dictionary
+        Contains variables ``lon``, ``lat``.
+
+    Returns
+    -------
+    locstream : ESMF.LocStream object
+
+    '''
+
+    lon = np.asarray(ds['lon'])
+    lat = np.asarray(ds['lat'])
+
+    if len(lon.shape) > 2:
+        raise ValueError("lon can only be 1d or 2d")
+    if len(lat.shape) > 2:
+        raise ValueError("lat can only be 1d or 2d")
+
+    if len(lon.shape) == 2:
+        lon = lon.flatten()
+    if len(lat.shape) == 2:
+        lat = lat.flatten()
```

Comment:
Better require strict 1-D shape which is the definition of locstream.

Reply:
I'm fine with it! maybe the flexibility is not a good thing here actually.